src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -1,35 +1,81 @@
+using System.Runtime.InteropServices;
+
 namespace Workleap.OpenApi.MSBuild;
 
 internal sealed class OasdiffManager : IOasdiffManager
 {
     private const string OasdiffVersion = "1.9.2";
 
+    private readonly ILoggerWrapper _loggerWrapper;
     private readonly IHttpClientWrapper _httpClientWrapper;
     private readonly IProcessWrapper _processWrapper;
-    private readonly string _toolDirectory;
+    private readonly string _openApiToolsDirectory;
+    private readonly string _oasdiffDirectory;
 
-    public OasdiffManager(IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
+    public OasdiffManager(ILoggerWrapper loggerWrapper, IProcessWrapper processWrapper, string openApiToolsDirectoryPath, IHttpClientWrapper httpClientWrapper)
     {
+        this._loggerWrapper = loggerWrapper;
         this._httpClientWrapper = httpClientWrapper;
         this._processWrapper = processWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._oasdiffDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
     }
 
+    private bool SkipRun { get; set; }
```

Comment:
I think this isn't necessary. It's set to true once when we catch and rethrow an exception.

Then it's bool checked when executing the tool, but because we threw an exception we will never reach the execution of the tool, right?